### PR TITLE
Update electronite

### PR DESCRIPTION
**Pull request by da1nerd**
#### Describe what your pull request addresses (Please include relevant issue numbers):

This updates electronite to 12.0.5 and corrects a couple of breaking changes that were introduced as a result.
See the related task in clickup https://app.clickup.com/t/g0x9gz.
The electronite release is at https://github.com/unfoldingWord-dev/electronite/releases/tag/v12.0.5-beta-graphite

#### Please include detailed Test instructions for your pull request:

Start the app and open the developer tools. You'll be able to see the correct electron version in the console.
image

If you would like to test that graphite is still properly enabled you can open a project that contains some Arabic text and visually confirm it is rendering correctly as seen below. (I copied the sample text into the verse below from the [SIL demo site](https://scripts.sil.org/cms/scripts/page.php?site_id=projects&item_id=graphite_fontdemo)).

> Keep in mind that Arabic text is read RTL.

image

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword/translationcore/7149)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -78,7 +78,7 @@
     "cross-env": "7.0.2",
     "css-loader": "3.4.2",
     "electron-devtools-installer": "3.0.0",
-    "electronite": "7.2.1-graphite",
+    "electronite": "^12.0.5-beta",
```
Comment (mannycolon) [inline package.json]: Should we add the beta version?

**Review by mannycolon — COMMENTED** [review]
Hey @da1nerd good job, there seems to be something wrong with the windows builds.

File: package.json (hunk 1 above)
Comment (da1nerd) [reply package.json]: that was an accident. I published 12.0.5 to npm but there was a bug in it. And NPM won't allow you to re-use a deployed version. So it has to be `12.0.5-[something]`. I could deploy it again as `12.0.5-graphite` so it seems more official if you think that is necessary.

**Comment by da1nerd** [comment]
> Hey @da1nerd good job, there seems to be something wrong with the windows builds.

Could you elaborate? I tested the windows builds before I uploaded them. They were compiled and tested in a VM with windows 10, but I don't think the VM aspect would affect it.

## Edit

I just realized you were talking about the windows translationCore builds @mannycolon . I forgot we had that pipeline :laughing: 
I'll investigate.

**Comment by da1nerd** [comment]
Looks like the windows builds have been failing for awhile possibly since last October.
See this one from Feb 8th https://github.com/unfoldingWord/translationCore/runs/1855056779

The logs from October have expired, but I'm guessing this is when things started breaking. https://github.com/unfoldingWord/translationCore/commit/4ce58c1db55b2f64d71031f0e5ddcce93d6d6ec1
A lot of changes were made to the github actions workflow.

**Comment by da1nerd** [comment]
I realized that I forked from master instead of develop so I'l rebase it

**Comment by da1nerd** [comment]
It took forever to figure this out, but you need to specify the shell when you programmatically set environment variables in windows.
By default it's `cmd` but I had to set it to `bash` in order for the variables to be set properly.

**Comment by da1nerd** [comment]
@mannycolon I fixed the windows builds and updated the version number so it doesn't include a "beta" tag on it. Please take another look once these builds finish. Thnks.

**Comment by mannycolon** [comment]
Great find @da1nerd

**Review by mannycolon — DISMISSED** [review]
Great job, just pending builds.

**Comment by mannycolon** [comment]
Testing branch & mac build locally

**Comment by mannycolon** [comment]
@da1nerd Should we worry about the following warnings on the terminal?

Screen Shot 2021-05-07 at 3 56 03 PM (2)
Screen Shot 2021-05-07 at 3 56 03 PM (2)

**Comment by mannycolon** [comment]
@da1nerd Should we also worry about this console warning?
Screen Shot 2021-05-07 at 3 58 35 PM

**Comment by da1nerd** [comment]
@mannycolon I think [this one](https://github.com/unfoldingWord/translationCore/pull/7149#issuecomment-834737343) is related to a third party dependency because that electron API method isn't used in the tC code. 

The warning about the [deprecated remote API](https://github.com/unfoldingWord/translationCore/pull/7149#issuecomment-834737984)  is not critical, in-fact if you take a look at [this change](https://github.com/unfoldingWord/translationCore/pull/7149/files#diff-ed330984cf54f0eebee9bc0784d5712ed8e70b056575e99016be7eb1b0817d5dR57) in the PR, I've specifically re-enabled the API because I didn't want to take too much time refactoring code if it wasn't necessary.

Both of those are for deprecated features that will be removed in future versions of Electron. They will continue to function properly until that time.

**Comment by da1nerd** [comment]
@mannycolon I didn't notice it at first because you circled two warnings above. One of those is because the developer tools extension seems to not be supported in the latest version of Electron. That is something that would be worth investigating.

## EDIT:
Looks like the React and Redux DevTools are not supported in the latest version of electron. https://github.com/electron/electron/issues/23662
Since translationCore is in maintenance mode this shouldn't be that big of a problem.

**Comment by mannycolon** [comment]
Good job @da1nerd, thanks for taking a look at that.

**Review by mannycolon — APPROVED** [review]

**Comment by mannycolon** [comment]
LGTM

**Pull request merged**